app/helpers/sign_up_sheet_helper.rb

```diff
@@ -55,25 +55,33 @@ def get_topic_bg_color(topic, max_team_size)
     'rgb(' + red + ',' + green + ',0)'
   end
 
-  # Render the participant info for a topic and assignment.
-  def render_participant_info(topic, assignment, participants)
+  # Show the participants present under a particular topic
+  def generate_html_for_participants(topic, assignment, participants, is_waitlisted)
     html = ''
     if participants.present?
-      chooser_present = false
       participants.each do |participant|
         next unless topic.id == participant.topic_id
 
-        chooser_present = true
         html += participant.user_name_placeholder
         if assignment.max_team_size > 1
           html += '<a href="/sign_up_sheet/delete_signup_as_instructor/' + participant.team_id.to_s + '?topic_id=' + topic.id.to_s + '"">'
           html += '<img border="0" align="middle" src="/assets/delete_icon.png" title="Drop Student"></a>'
         end
-        html += '<font color="red">(waitlisted)</font>' if participant.is_waitlisted
+        html += '<font color="red">(waitlisted)</font>' if is_waitlisted
```

Comment:
Helpers, in my opinion, should be exclusively used for getting the data, and all the html rendering should be done in the `.html.erb` files. Please refactor this as well.